WireDomain/Sources/WireDomain/Repositories/Connections/ConnectionsLocalStore.swift

```diff
@@ -16,15 +16,16 @@
 // along with this program. If not, see http://www.gnu.org/licenses/.
 //
 
-import CoreData
-import Foundation
-import WireAPI
 import WireDataModel
 
-protocol ConnectionsLocalStoreProtocol {
+// sourcery: AutoMockable
+public protocol ConnectionsLocalStoreProtocol {
 
+    /// Save connection and related objects to local storage.
+    /// - Parameter connectionInfo: connection object
+    
     func storeConnection(
-        _ connectionPayload: Connection
+        _ connectionInfo: ConnectionInfo
```

Comment:
You'll see changes like this in this PR: storage layer should know nothing about API objects so we create a passthrough model between repository and local store.

1. Repo will fetch API model from remote.
2. Repo will prepare data for the local store by mapping it to a domain model
3. Repo will pass this model to the local store

Reply:
So now, it looks like all NSManagedObjects or context are isolated to Local stores, right?
I wonder if they would be cases where we want to be more flexible : thinking about the save contexts strategy we discussed.

Or going even further isolate all this to a WireStorage package

Reply:
yes most of the storage operations are isolated to the local stores. 

Moving to a WireStorage package could make sense as we used that approach for WireAPI that contains `UsersAPI` `ConversationsAPI`. WireStorage could follow that approach with `UsersLocalStore` `ConversationsLocalStore` and later introduce a generic storage service, just like we did with `APIService`.

Reply:
I think we should consider this still an open topic as it has a lot of implications to the overall architecture that should be further investigated. Having said that, I think that for now it is fine to introduce such isolation since it's generally easier to remove abstractions that it is to put them in place.

Reply:
@johnxnguyen you mean the WireStorage move ?

Reply:
resolving this conversation to merge this PR, I will follow up on that discussion on our dedicated refactoring group